### PR TITLE
Inner silence uat interim computing method [PROTO]

### PR DESCRIPTION
**Pull request by metas-rc**

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/invoicecandidate/FlatrateTerm_Handler.java [hunk 1]
```diff
@@ -203,7 +203,7 @@ public void setOrderedData(@NonNull final I_C_Invoice_Candidate ic)
 		ic.setQtyEntered(calculateQtyOrdered.toBigDecimal());
 		ic.setC_UOM_ID(calculateQtyOrdered.getUomId().getRepoId());
 
-		final ProductId productId = ProductId.optionalOfRepoId(ic.getM_Product_ID())
+		final ProductId productId = ProductId.optionalOfRepoId(term.getM_Product_ID())
```
Comment (teosarca) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/invoicecandidate/FlatrateTerm_Handler.java:206]: why?

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java [hunk 2]
```diff
@@ -283,7 +283,16 @@ private ComputingResponse getComputingMethodResponse(@NonNull final CreateInvoic
...
+		final UomId stockUomId;
+
+		if(moduleConfig.getModularContractType().isMatching(ComputingMethodType.INTERIM_CONTRACT))
+		{
+			stockUomId = productBL.getStockUOMId(modularContract.getM_Product_ID());
+		}
+		else
+		{
+			stockUomId = productBL.getStockUOMId(moduleConfig.getProductId());
+		}
```
Comment (teosarca) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java:295]: why do we need this validation?

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/invoicecandidate/FlatrateTerm_Handler.java (hunk 1 above)
Comment (metas-rc) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/invoicecandidate/FlatrateTerm_Handler.java:206]: It was a typo introduced here https://github.com/metasfresh/metasfresh/commit/7cb4f03ae9ef1e8041ba9d732edb9cfc1f727249#diff-4d8ffe171e8889a74796e34c9a5456fa5debf755d14f2cc3f57d2e059007ca2bR206  and I fixed it

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java:297]: Added a comment in the code about validating the UOM of the contract product if interim.

**Pull request merged**